torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp

```diff
@@ -70,7 +70,11 @@ bool isWeight(Module& module, Value* v) {
 
 Node* insertChooseQParams(Graph* graph, Value* original_val) {
   std::string choose_qparams_func = "_choose_qparams_per_tensor";
-  auto reduce_range = graph->insertConstant(false);
+  bool reduce_range_param = false;
+  if (at::globalContext().qEngine() == at::QEngine::FBGEMM) {
```

Comment:
this means we have an implicit argument of qengine, which makes the produced quantized model tied to fbgemm, which is not desirable, is there any plan to remove this in the future?

Reply:
This argument should not affect the final converted quantized model. The `reduce_range` option gets added to `linear_dynamic` op as an argument. During runtime if backend is not fbgemm, it gets ignored.
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/quantized/cpu/qlinear_dynamic.cpp#L20
and https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/quantized/cpu/qlinear_dynamic.cpp#L222

Reply:
what about the other way around? if we produce the model with qengine set to qnnpack, does it run on fbgemm qengine?

Reply:
It will run, but it might not be the most optimal. In the future we can make reduce_range default to true for the op, but that might break BC for numerics of existing models.